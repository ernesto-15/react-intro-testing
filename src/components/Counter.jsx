import React, { useState } from 'react';
import PropTypes from 'prop-types';

const Counter = ({ value }) => {
  const [count, setCount] = useState(value);

  const handleAdd = () => {
    setCount((count) => count + 1);
  };

  const handleSubtract = () => {
    setCount((count) => count - 1);
  };

  const handleReset = () => {
    setCount(value);
  };

  return (
    <>
      <h1>Counter App</h1>
      <p>{count}</p>
      <button onClick={handleAdd}>+</button>
      <button onClick={handleSubtract}>-</button>
      <button onClick={handleReset}>Reset</button>
    </>
  );
};

Counter.propTypes = {
  value: PropTypes.number,
};

export default Counter;
