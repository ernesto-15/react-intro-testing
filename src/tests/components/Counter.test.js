import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import Counter from '../../components/Counter';

describe('Tests in Counter', () => {
  let wrapper = shallow(<Counter />);

  beforeEach(() => {
    wrapper = shallow(<Counter />);
  });

  test('should show Counter component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should show value prop', () => {
    const value = 100;
    const wrapper = shallow(<Counter value={value} />);
    const valueText = Number(wrapper.find('p').text());
    expect(valueText).toBe(value);
  });

  test('should add with + button', () => {
    wrapper.find('button').at(0).simulate('click');
    const valueText = Number(wrapper.find('p').text());
    expect(valueText).toBe(16);
  });

  test('should add with - button', () => {
    wrapper.find('button').at(1).simulate('click');
    const valueText = Number(wrapper.find('p').text());
    expect(valueText).toBe(14);
  });

  test('should add with reset button', () => {
    const value = 100;
    const wrapper = shallow(<Counter value={value} />);
    wrapper.find('button').at(1).simulate('click');
    wrapper.find('button').at(2).simulate('click');
    const valueText = Number(wrapper.find('p').text());
    console.log(valueText);
    expect(valueText).toBe(value);
  });
});
