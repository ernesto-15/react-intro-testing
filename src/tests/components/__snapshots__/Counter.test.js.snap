// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in Counter should show Counter component 1`] = `
<Fragment>
  <h1>
    Counter App
  </h1>
  <p>
    15
  </p>
  <button
    onClick={[Function]}
  >
    ++
  </button>
  <button
    onClick={[Function]}
  >
    --
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
</Fragment>
`;
